stat: ROC return FPR and TPR with NaN values

With #404 merged, TestROC fails:

```
--- FAIL: TestROC (0.00s)
	roc_test.go:175: 12: unexpected FPR got:[NaN NaN 1] want:[0 0 1]
	roc_test.go:175: 13: unexpected FPR got:[NaN 1] want:[0 1]
	roc_test.go:175: 14: unexpected FPR got:[NaN NaN NaN NaN NaN NaN 1] want:[0 0 0 0 0 0 1]
	roc_test.go:175: 15: unexpected FPR got:[NaN 1] want:[0 1]
	roc_test.go:172: 17: unexpected TPR got:[NaN 1] want:[0 1]
```

--- a/stat/roc.go
+++ b/stat/roc.go
@@ -108,8 +108,13 @@ func ROC(n int, y []float64, classes []bool, weights []float64) (tpr, fpr []floa
 		fpr = fpr[:(bin + 1)]
 	}
 
-	invNeg := 1 / nNeg
-	invPos := 1 / nPos
+	var invNeg, invPos float64
+	if nNeg != 0 {
+		invNeg = 1 / nNeg
+	}
+	if nPos != 0 {
+		invPos = 1 / nPos
+	}
 	for i := range tpr {
 		tpr[i] *= invPos
 		fpr[i] *= invNeg

